src/services/YotiCallbackProcessor.ts

```diff
@@ -13,135 +13,171 @@ import { ServicesEnum } from "../models/enums/ServicesEnum";
 import { KmsJwtAdapter } from "../utils/KmsJwtAdapter";
 import { AuthSessionState } from "../models/enums/AuthSessionState";
+import { GenerateVerifiableCredential } from "./GenerateVerifiableCredential";
 
 export class YotiCallbackProcessor {
 
-  	private static instance: YotiCallbackProcessor;
-
-  	private readonly logger: Logger;
+  private static instance: YotiCallbackProcessor;
+	
+  private readonly logger: Logger;
+	
+  private readonly metrics: Metrics;
+
+  private readonly yotiService: YotiService;
+
+  private readonly f2fService: F2fService;
+
+  private readonly environmentVariables: EnvironmentVariables;
+
+  private readonly verifiableCredentialService: VerifiableCredentialService;
+
+  private readonly kmsJwtAdapter: KmsJwtAdapter;
+
+  private readonly generateVerifiableCredential: GenerateVerifiableCredential;
+
+  constructor(
+  	logger: Logger,
+  	metrics: Metrics,
+  	YOTI_PRIVATE_KEY: string,
+  ) {
+  	this.logger = logger;
+  	this.metrics = metrics;
+  	this.environmentVariables = new EnvironmentVariables(logger, ServicesEnum.CALLBACK_SERVICE);
+  	this.yotiService = YotiService.getInstance(this.logger, this.environmentVariables.yotiSdk(), this.environmentVariables.resourcesTtlInSeconds(), this.environmentVariables.clientSessionTokenTtlInSeconds(), YOTI_PRIVATE_KEY, this.environmentVariables.yotiBaseUrl());
+  	this.f2fService = F2fService.getInstance(this.environmentVariables.sessionTable(), this.logger, createDynamoDbClient());
+  	this.kmsJwtAdapter = new KmsJwtAdapter(this.environmentVariables.kmsKeyArn());
+  	this.verifiableCredentialService = VerifiableCredentialService.getInstance(this.environmentVariables.sessionTable(), this.kmsJwtAdapter, this.environmentVariables.issuer(), this.logger);
+  	this.generateVerifiableCredential = GenerateVerifiableCredential.getInstance(this.logger);
+  }
+
+  static getInstance(
+  	logger: Logger,
+  	metrics: Metrics,
+  	YOTI_PRIVATE_KEY: string,
+  ): YotiCallbackProcessor {
+  	if (!YotiCallbackProcessor.instance) {
+  		YotiCallbackProcessor.instance = new YotiCallbackProcessor(
+  			logger,
+  			metrics,
+  			YOTI_PRIVATE_KEY,
+  		);
+  	}
+  	return YotiCallbackProcessor.instance;
+  }
 
-  	private readonly metrics: Metrics;
+  async processRequest(eventBody: any): Promise<Response> {
+  	const yotiSessionID = eventBody.session_id;
 
-  	private readonly yotiService: YotiService;
+  	this.logger.info({ message: "Fetching status for Yoti SessionID" }, { yotiSessionID });
+  	const completedYotiSessionInfo = await this.yotiService.getCompletedSessionInfo(yotiSessionID);
 
-  	private readonly f2fService: F2fService;
+  	if (!completedYotiSessionInfo) {
+  		this.logger.error({ message: "No YOTI Session found with ID:" }, { yotiSessionID });
+  		throw new AppError(HttpCodesEnum.SERVER_ERROR, "Yoti Session not found");
+  	}
 
-		private readonly environmentVariables: EnvironmentVariables;
+  	if (completedYotiSessionInfo.state !== "COMPLETED") {
```

Comment:
Should we move the COMPLETED state to an ENUM if we have any other states else move it to constants?